custom_components/e3dc_rscp/coordinator.py

```diff
@@ -448,13 +513,31 @@ async def async_manual_charge(self, charge_amount: int) -> None:
             _LOGGER.debug("Successfully started manual charging")
 
 
-def create_e3dcinstance(username: str, password: str, host: str, rscpkey: str) -> E3DC:
+def create_e3dcinstance(
+    username: str, password: str, host: str, rscpkey: str, config: dict = None
+) -> E3DC:
     """Create the actual E3DC instance, this will try to connect and authenticate."""
-    e3dc = E3DC(
-        E3DC.CONNECT_LOCAL,
-        username=username,
-        password=password,
-        ipAddress=host,
-        key=rscpkey,
-    )
+    if config is None:
```

Comment:
You can pass an empty dictionary to pye3dc, that'll achive the same effect but will remove the if/else, simplifying the code. It should be enough if you replace `dict = None` with an empty dictionary as default value, drop the if, that should work. 

This is the library code parsing the argument:

```python
        if "configuration" in kwargs:
            configuration = kwargs["configuration"]
            if "pvis" in configuration and isinstance(configuration["pvis"], list):
                self.pvis = configuration["pvis"]
            if "powermeters" in configuration and isinstance(
                configuration["powermeters"], list
            ):
                self.powermeters = configuration["powermeters"]
            if "batteries" in configuration and isinstance(
                configuration["batteries"], list
            ):
                self.batteries = configuration["batteries"]
```

Reply:
Please simplify the code here in this style (avoiding duplication):

```python
def create_e3dcinstance(...)
     if config is None:
        config = {}
    # Use the code in the else branch from here on only
```